### PR TITLE
Rs logging sdk

### PR DESCRIPTION
**Pull request by rsnm2**
Draft

To finalize with code landed yesterday with input from Damian

### CODE REVIEW

File: examples/logging/README.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
+# DeepSparse Logging
+
+DeepSparse Logging provides operational teams with access to the telemetry needed to monitor a deployment. 
+
+For users seeking to put ML into production, these data are the raw materials that underpin the monitoring processes needed to create a system with consistently fresh and accurate predictions.
```
Comment (dbogunowicz) [inline examples/logging/README.md:5]: Nit: do not like the "raw materials" wording here. Feels odd. Maybe the "essential source of truth" or smth?

File: examples/logging/README.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+
+<p align="center">
+     <img src="continual-learning.png"
+          alt="Continual Learning Diagram"
+          width="50%"
+     />
+</p>
+
+There are many types of downstream monitoring tasks that ML teams may want to perform. Some are easier to do (such as looking at system performance) and some are harder (looking at model accuracy requires manually labeling some data afterwards). Examples include:
```
Comment (dbogunowicz) [inline examples/logging/README.md:14]: Nit:
```suggestion
There are many types of downstream monitoring tasks that ML teams may want to perform. The difficulty of the tasks varies from relatively easy (simple system performance analysis ) to challenging (assessing the accuracy of the system in the wild by manually labeling the input data distribution post-factum).  Examples include:
```

File: examples/logging/README.md [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+- **System performance:** what is the latency/throughput of a query?
+- **Data quality:** is there an error in the model pipeline?
```
Comment (dbogunowicz) [inline examples/logging/README.md:16]: I don't think the error in the model pipeline is a data quality problem. What do we want to say here?

File: examples/logging/README.md [hunk 4]
```diff
@@ -0,0 +1,340 @@
...
+- **Data distribution shift:** are the inputs and outputs from the system expected?
```
Comment (dbogunowicz) [inline examples/logging/README.md:17]:
```suggestion
- **Data distribution shift:** does the input data distribution deviates over time to the point where the model stops to deliver reliable predictions?
```

File: examples/logging/README.md [hunk 5]
```diff
@@ -0,0 +1,340 @@
...
+- **Model accuracy:** what is the accuracy of the predictions vs human-labeled?
```
Comment (dbogunowicz) [inline examples/logging/README.md:18]: Maybe more broadly:
```suggestion
- **Model accuracy:** percentage of correct predictions that a model achieves?
```

File: examples/logging/README.md [hunk 6]
```diff
@@ -0,0 +1,340 @@
...
+
+DeepSparse Logging is designed to provide maximum flexibility for users to extract whatever data is needed from a production inference pipeline into the logging system of their choice.
```
Comment (dbogunowicz) [inline examples/logging/README.md:20]: Very important and powerful sentence, me gusta mucho.

File: examples/logging/README.md [hunk 7]
```diff
@@ -0,0 +1,340 @@
...
+- **System performance:** what is the latency/throughput of a query?
+- **Data quality:** is there an error in the model pipeline?
+- **Data distribution shift:** are the inputs and outputs from the system expected?
+- **Model accuracy:** what is the accuracy of the predictions vs human-labeled?
+
+DeepSparse Logging is designed to provide maximum flexibility for users to extract whatever data is needed from a production inference pipeline into the logging system of their choice. 
+
+## Metrics 
+DeepSparse Logging provides access to two types of metrics:
+- **System Logging Metrics** give operations teams access to granual performance metrics, diagnosing and isolating deployment system health. Examples include CPU utilization and query latency.
```
Comment (dbogunowicz) [inline examples/logging/README.md:24]:
```suggestion
- **System Logging Metrics** give MLOps teams access to granular performance metrics for quick and efficient diagnosis of deployment system health. Examples may include CPU utilization and query latency.
```

File: examples/logging/README.md [hunk 8]
```diff
@@ -0,0 +1,340 @@
...
+     - See [below - UPDATE LINK](system-logging-metrics) for a full list of available metrics.
+
+- **Data Logging Metrics** give ML teams access to data at each stage of an ML pipeline, supporting downsteam tasks like measuring accuracy and data drift. Examples include raw inputs and projections thereof such as mean pixel value.
```
Comment (dbogunowicz) [inline examples/logging/README.md:27]:
```suggestion
- **Data Logging Metrics** give ML teams access to the data that flows through each stage of an ML pipeline. This facilitates inspection of the data, understanding of its properties, detecting edge cases and possible data drift. Examples include logging raw data distribution, projections thereof (e.g. mean pixel value of an image batch), and features distribution.
```

File: examples/logging/README.md [hunk 9]
```diff
@@ -0,0 +1,340 @@
...
+     - See [below - UPDATE LINK](data-logging-functions) for a full list of built-in functions and how to create a custom function
+
+## Configuration
+DeepSparse Logging is configured via YAML files.
+
+**System Logging** is *enabled* by default and all metrics are pre-defined. It can be disabled 
+globally or at the group level by adding the key-value pairs with `on` or `off` status.
```
Comment (dbogunowicz) [inline examples/logging/README.md:34]: Caution -> this is not something that we have implemented for 1.3.0. Can be cherry-picked easily though @bfineran

File: examples/logging/README.md [hunk 10]
```diff
@@ -0,0 +1,340 @@
...
+        
+**Data Logging** is *disabled* by default. A YAML configuration file is used to specify which data or functions thereof to log.
```
Comment (dbogunowicz) [inline examples/logging/README.md:36]: We start talking about the functions here, I think it is appropriate to have introduced the metric functions at this point.

File: examples/logging/README.md [hunk 11]
```diff
@@ -0,0 +1,340 @@
...
+     - See [below - UPDATE LINK](data-logging-functions) for a full list of built-in functions and how to create a custom function
+
+## Configuration
+DeepSparse Logging is configured via YAML files.
+
+**System Logging** is *enabled* by default and all metrics are pre-defined. It can be disabled 
+globally or at the group level by adding the key-value pairs with `on` or `off` status.
+        
+**Data Logging** is *disabled* by default. A YAML configuration file is used to specify which data or functions thereof to log. 
+There are 4 `stages` in the inference pipeline where Data Logging can occur:
```
Comment (dbogunowicz) [inline examples/logging/README.md:37]: By default. Maybe let's add that the user can overrule this convention.

File: examples/logging/README.md [hunk 12]
```diff
@@ -0,0 +1,340 @@
...
+
+````
+stages: pipeline_inputs >> engine_inputs >> engine_outputs >> pipeline_outputs
```
Comment (dbogunowicz) [inline examples/logging/README.md:40]: Let's add a short paragraph where we explain what those data structures are.

File: examples/logging/README.md [hunk 13]
```diff
@@ -0,0 +1,340 @@
...
+
+# system_logging: on/off    # flag to turn all system logging on/off; defaults to on
+system_logging:
+    group: on/off           # flag to turn off a group; if omitted, defaults to on
+    
+data_logging:
+    stage:
+      mapping:
+        # first function
+        - func: builtins:foo            # [REQUIRED STR] function identifier  (built-in or path to custom)
```
Comment (dbogunowicz) [inline examples/logging/README.md:57]: We have decided that a `builtins` metric function name is to be passed as an argument to `func` without prepending anything (to reduce the noise).

```suggestion
        - func: foo            # [REQUIRED STR] function identifier  (built-in or path to custom)
```

File: examples/logging/README.md [hunk 14]
```diff
@@ -0,0 +1,340 @@
...
+**Data Logging** is *disabled* by default. A YAML configuration file is used to specify which data or functions thereof to log. 
+There are 4 `stages` in the inference pipeline where Data Logging can occur:
+
+````
+stages: pipeline_inputs >> engine_inputs >> engine_outputs >> pipeline_outputs
+````
+
+The following YAML format is used to specify the configuration:
+```yaml
+loggers:
```
Comment (dbogunowicz) [inline examples/logging/README.md:45]: I am sorry @rsnm2, there were several tiny, ad-hoc changes made to the structure of a yaml file. Let me quickly present them here using the yaml below as an example:
```yaml
loggers: # loggers are now a dictionary logger_name -> logger_arguments
  python: # loggers that do not take any arguments can either look like this or python: {}
  prometheus: # also `prometheus: ` or `prometheus: {}` is a valid entry if we want to use the default arguments 
    port: 6100
    
system_logging:
    group: on/off # as mentioned before, let's be cautious, this is not supported atm

data_logging:
    some_pipeline_1/pipeline_inputs.images: # "/" separate the consecutive parts of the target identifier, "." separates target identifier from the property
        # first function
        - func: foo # if `foo` is a built-in function, we skip the "built-ins:" prefix, only adds noise
          frequency: 1000               
          # if no `target_loggers` are specified, we log to all loggers                
        # second function
        - func: path/to/custom.py:bar # or np.mean, torch.mean etc.
          frequency: 10000
          target_loggers: 
            - prometheus # must be List[str], where str is the name of the logger
        # ... list of as many functions as desired
```

Comment (dbogunowicz) [reply examples/logging/README.md:45]: @rsnm2 let's not forget that if the target contains a property, we can further access properties attributes by slicing/indexing/dict access:
```yaml
data_logging:
    some_pipeline_1/pipeline_inputs.images[0:2][0, :-1]['key_3'][5]: # etc
```

File: examples/logging/README.md [hunk 15]
```diff
@@ -0,0 +1,340 @@
...
+        # ... list of as many functions as desired
+```
+
+<details>
+     <summary>Click for a tangible example YAML snippit</summary>
+
+```yaml
+loggers:
+    - prometheus
+    - custom_logger
```
Comment (dbogunowicz) [inline examples/logging/README.md:73]: To create a custom logger, it needs at least one argument: `path`. Please look at the PR description:
https://github.com/neuralmagic/deepsparse/pull/768

File: examples/logging/README.md [hunk 16]
```diff
@@ -0,0 +1,340 @@
...
+
+system_logging:
```
Comment (dbogunowicz) [inline examples/logging/README.md:75]: all of that not implemented :)

File: examples/logging/README.md [hunk 17]
```diff
@@ -0,0 +1,340 @@
...
+    deployment_details: off
+    request_details: off
+    prediction_latency: on 
+    dynamic_batch_latency: off
+    # resource_utilization: on      << omitted groups are turned on by default
+    
+data_logging:
+     pipeline_inputs:
+       mapping:
+         - func: builtins/max                        # built-in function
```
Comment (dbogunowicz) [inline examples/logging/README.md:85]: We need to remove `builtins` prefix and add a function name that actually exists (`max` doesn't):
https://github.com/neuralmagic/deepsparse/blob/main/src/deepsparse/loggers/metric_functions/built_ins.py

File: examples/logging/README.md [hunk 18]
```diff
@@ -0,0 +1,340 @@
...
+           frequency: 100                            # logs once per 100 predictions
+           target: prometheus                        # logs to prometheus
```
Comment (dbogunowicz) [inline examples/logging/README.md:87]:
```suggestion
```

File: examples/logging/README.md [hunk 19]
```diff
@@ -0,0 +1,340 @@
...
+           
+         - func: /path/to/logging_fns.py:my_fn       # custom function
+           # frequency:                              # not specified, defaults to once per 1000 predictions
+           # target:                                 # not specified, defaults to all loggers
```
Comment (dbogunowicz) [inline examples/logging/README.md:91]:
```suggestion
```

File: examples/logging/README.md [hunk 20]
```diff
@@ -0,0 +1,340 @@
...
+
+     engine_inputs:
+       mapping:
+         - func: builtins/channel-mean               # pre-defined function (logs per channel mean pixel)
```
Comment (dbogunowicz) [inline examples/logging/README.md:95]: some comments as before

File: examples/logging/README.md [hunk 21]
```diff
@@ -0,0 +1,340 @@
...
+    - custom_logger
+
+system_logging:
+    deployment_details: off
+    request_details: off
+    prediction_latency: on 
+    dynamic_batch_latency: off
+    # resource_utilization: on      << omitted groups are turned on by default
+    
+data_logging:
```
Comment (dbogunowicz) [inline examples/logging/README.md:82]: let's apply my previous remarks about the structure of `data_logging` also here.

File: examples/logging/README.md [hunk 22]
```diff
@@ -0,0 +1,340 @@
...
+- Engine Outputs + Pipeline Outputs: No logging occurs at this stage
+
+</details>
+
+## Loggers
+Logging to Prometheus works out-of-the-box and users can add custom loggers.
+
+### Prometheus Logger
+
+[@DAMIAN] - need your help here
```
Comment (dbogunowicz) [inline examples/logging/README.md:118]: Let's sync today on the goal of this paragraph and I will get to it asap.

File: examples/logging/README.md [hunk 23]
```diff
@@ -0,0 +1,340 @@
...
+
+### Custom Logger
+
+[@DAMIAN] - need your help here
```
Comment (dbogunowicz) [inline examples/logging/README.md:122]: Let's sync today on the goal of this paragraph and I will get to it asap.

File: examples/logging/README.md [hunk 24]
```diff
@@ -0,0 +1,340 @@
...
+       model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/pruned95_obs_quant-none
+       batch_size: 1
+       # data_logging:                          < nothing logged at this endpoint 
+```
+
+</details>
+
+### Pipeline Usage
+
+The `ManagerLogger` class handles logging and is passed as the `logger` argument to a `Pipeline`. `ManagerLogger` is initialized with the `config` argument, which is a path to a configuration file in the format described above.
```
Comment (dbogunowicz) [inline examples/logging/README.md:193]: Warning: `ManagerLogger` does not exist (called `MultiLogger`).

File: examples/logging/README.md [hunk 25]
```diff
@@ -0,0 +1,340 @@
...
+For example, with the QA pipeline:
+
+```python
+from deepsparse import Pipeline
+
+# SparseZoo model stub or path to ONNX file
+model_path = "zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/12layer_pruned80_quant-none-vnni"
+
+# logger object referencing the local logging config file
+logger = ManagerLogger(config="pipeline-logging-config.yaml")
```
Comment (dbogunowicz) [inline examples/logging/README.md:204]: This is not supported atm.
In theory, any logger that they provide should work, not only the `Manager(Multi)Logger`.

I do not think that we should allow here the users to specify the loggers, which would later be directly fed to the pipeline. Maybe for simple use cases, they could directly plug here a `PythonLogger`. 

But for good performance / real-life usefulness, they'd need some logger composition, which should be handled by some helper function `logger_from_config(config: str)`

@bfineran

File: examples/logging/README.md [hunk 26]
```diff
@@ -0,0 +1,340 @@
...
+        - func: buildins:identity
+          frequency: 1000
+          target: custom_logger
+          
+     # - engine_outputs           < not specified, so nothing logged at this target
+     # - pipeline_outputs         < not specified, so nothing logged at this target
+```
+</details>
+
+If no `ManagerLogger` is passed to a `Pipeline`, then logging is disabled.
```
Comment (dbogunowicz) [inline examples/logging/README.md:254]:
```suggestion
If no logger is passed to a `Pipeline`, then logging is disabled.
```

File: examples/logging/README.md [hunk 27]
```diff
@@ -0,0 +1,340 @@
...
+## Tutorials
+There are serveral examples integrating Prometheus and DeepSparse logging available:
+- [Server with Prometheus / Grafana](https://github.com/neuralmagic/deepsparse/tree/rs-logging-sdk/logging-sdk/tutorial-server-prometheus)
+- [Pipeline with Prometheus / Grafana](https://github.com/neuralmagic/deepsparse/tree/rs-logging-sdk/logging-sdk/tutorial-pipeline-prometheus)
+- [Monitoring while running Kubernetes](https://github.com/neuralmagic/deepsparse/tree/rs-logging-sdk/logging-sdk/tutorial-kubernetes-prometheus)
+- [Custom Logger (S3 Bucket)](https://github.com/neuralmagic/deepsparse/tree/rs-logging-sdk/logging-sdk/tutorial-custom-logger)
+
+## System Logging Metrics
+
+**TO BE UPDATED** ONCE WE HAVE THE LIST OF METRICS FOR 1.3
```
Comment (dbogunowicz) [inline examples/logging/README.md:265]: Afaik only latencies right?

File: examples/logging/README.md [hunk 28]
```diff
@@ -0,0 +1,340 @@
...
+The following metric groups are available:
+- Prediction Latency (`prediction_latency`)
+     
+### Metrics
+The following metrics are logged for each group:
+
+#### Prediction Latency
+Metric           |Metric Name              |Description                              |Granularity    |Usage  |Frequency      |
+|---------------- |-------------------------|-----------------------------------------|---------------|-------|---------------|
+|Total Time       |`sl_pl_total_time`       |End-to-end prediction time               |Per Pipeline   |All    |Per Prediction |
```
Comment (dbogunowicz) [inline examples/logging/README.md:277]: what's `sl_pl`?

File: examples/logging/README.md [hunk 29]
```diff
@@ -0,0 +1,340 @@
...
+|Preprocess Time  |`sl_pl_preprocess_time`  |Time spent in pre-processing step        |Per Pipeline   |All    |Per Prediction |
+|Engine Time      |`sl_pl_engine_time`      |Time spent in engine forward pass        |Per Pipeline   |All    |Per Prediction |
+|Postprocess Time |`sl_pl_postprocess_time` |Time spent in post-processing step       |Per Pipeline   |All    |Per Prediction |
+
+
+##  Data Logging Functions
+
+**TO BE UPDATED** ONCE WE HAVE LIST FOR 1.3
```
Comment (dbogunowicz) [inline examples/logging/README.md:285]: https://github.com/neuralmagic/deepsparse/blob/main/src/deepsparse/loggers/metric_functions/built_ins.py

File: examples/logging/README.md [hunk 30]
```diff
@@ -0,0 +1,340 @@
...
+
+Built-in functions are predefined in DeepSparse. 
+   
+|Function Name  |Description                        |
+|---------------|-----------------------------------|
+|`mean_pixel_1` |Mean pixel value in first channel  |
+|`mean_pixel_2` |Mean pixel value in second channel |
+|`mean_pixel_3` |Mean pixel value in third channel  |
+
+They are applied in the form `func: builtins/function_name`. The YAML file would look as follows:
```
Comment (dbogunowicz) [inline examples/logging/README.md:297]:
```suggestion
They are applied in the form `func: function_name`. The YAML file would look as follows:
```

File: examples/logging/README.md [hunk 31]
```diff
@@ -0,0 +1,340 @@
...
+  
+loggers:
+  prometheus:
+    port:5555
+  
+data_logging:
+  pipeline_inputs:
+    mapping:
+      - func: builtin:function
+      - frequency: 1000
```
Comment (dbogunowicz) [inline examples/logging/README.md:309]:
```suggestion
        frequency: 1000
```

File: examples/logging/README.md [hunk 32]
```diff
@@ -0,0 +1,340 @@
...
+      - target: prometheus
```
Comment (dbogunowicz) [inline examples/logging/README.md:310]:
```suggestion
```

File: examples/logging/tutorial-custom-logger/README.md [hunk 33]
```diff
@@ -0,0 +1,31 @@
+# Custom Logger
+
+This page explains how to create a Custom Logger for use with DeepSparse Logging.
+
+### Why Might I Need A Custom Logger?
```
Comment (dbogunowicz) [inline examples/logging/tutorial-custom-logger/README.md:5]: Do we need to motivate the need for a custom logger through the lack of features on the Prometheus side? I'd say the sufficient motivation is user empowerment to log anything anywhere

File: examples/logging/tutorial-custom-logger/README.md [hunk 34]
```diff
@@ -0,0 +1,31 @@
...
+- Collecting a dataset for retraining the model with fresh data
+
+Additionally, there are many MLOps projects/companies focused on Model Monitoring, which need the information from the inference 
+pipelines to detect data drift. Custom Loggers enable you to connect DeepSparse deployment to these tools.
+
+## How To Use The Custom Logger
+
+Custom Loggers should inherit from the `ManagerLogger` class. 
+
+[@DAMIAN please describe what to do]
```
Comment (dbogunowicz) [inline examples/logging/tutorial-custom-logger/README.md:20]: Aye aye capt'n.

File: examples/logging/tutorial-kubernetes-prometheus/README.md [hunk 35]
```diff
@@ -0,0 +1,261 @@
+# DeepSparse + Kubernetes + Prometheus/Grafana
+
+Since DeepSparse runs on CPUs, you can scale deployments elastically with Kubernetes just like any other workload.
```
Comment (dbogunowicz) [inline examples/logging/tutorial-kubernetes-prometheus/README.md:3]: I do not see the causality here between running on CPU and scaling.

File: examples/logging/tutorial-kubernetes-prometheus/README.md [hunk 36]
```diff
@@ -0,0 +1,261 @@
...
+
+This tutorial demonstrates how to monitor a DeepSparse in a Kubernetes cluster with Prometheus.
```
Comment (dbogunowicz) [inline examples/logging/tutorial-kubernetes-prometheus/README.md:5]: I would love to have some testing of this example in-place if possible...

File: examples/logging/tutorial-server-prometheus/README.md [hunk 37]
```diff
@@ -0,0 +1,157 @@
...
+
+You may visit `localhost:9090` to inspect whether Prometheus recognizes the `metrics` endpoint (`Status` -> `Targets`)
+
+![img.png](images/img_1.png)
+
+Visit `localhost:3000` to launch Grafana. Log in with the default username (`admin`) and password (`admin`). 
+Setup the Prometheus data source (`Add your first data source` -> `Prometheus`). 
+
+Now you should be ready to create/import your dashboard. If you decide to import the dashboard, either upload `grafana/dashboard.json` or 
+paste its contents using Grafana's `import` functionality.
```
Comment (dbogunowicz) [inline examples/logging/tutorial-server-prometheus/README.md:155]: Let's test whether this is going to work for the user off-the bat. I might have some dashboard settings cached when I used import, so this did not reflect the "clean slate" setup that the user will be coming with.

File: examples/logging/tutorial-server-prometheus/README.md [hunk 38]
```diff
@@ -0,0 +1,157 @@
...
+
+![img.png](images/img_2.png)
```
Comment (dbogunowicz) [inline examples/logging/tutorial-server-prometheus/README.md:157]: Do we want to make those images nicer for the official documentation?

File: examples/logging/tutorial-server-prometheus/README.md [hunk 39]
```diff
@@ -0,0 +1,157 @@
...
+
+## 3. Launch the Python Client and Run Inference
+
+`client.py` is a simple client, that periodically sends requests to the Server. The client simulates the behavior of some application, 
+that sends the raw inputs to the inference server and receives the outputs. 
+
+Run the following to start inference:
+
+```bash
+python client/client.py client/piglet.jpg 5543
```
Comment (dbogunowicz) [inline examples/logging/tutorial-server-prometheus/README.md:141]: This was just some stupid image from the internet, just saying :) Maybe let's make sure we do not infringe any copyrights.

File: examples/logging/tutorial-server-prometheus/README.md [hunk 40]
```diff
@@ -0,0 +1,157 @@
+# DeepSparse Server + Prometheus/Grafana
```
Comment (dbogunowicz) [inline examples/logging/tutorial-server-prometheus/README.md:1]: This is a duplicate of the other tutorial README.md right? This one should be way more complex (prometheus operator setup etc.) :)

@rsnm2 I propose the following.
This tutorial is a bit of a big boi. Also, Mark expressed his interest in having here some more details on how the users can pull the system metrics directly from the operator.
Let's not try to bite off too large of a piece. I propose to hold off with this tutorial for now and focus on the documentation. Once this is in a good shape, we can come back to tutorials, freshen them up, put some lipstick and test them. I am just not sure about the deadlines, we can sync on that.

**Review by dbogunowicz — COMMENTED** [review]
Great start! But yeah, we probably need to do some more work:
- making sure that the docs reflects the current state of the codebase
- covering the full landscape of functionalities in the docs
- polishing up the tutorials (I propose we proritize the first two items before that one).

File: examples/logging/README.md (hunk 3 above)
Comment (rsnm2) [reply examples/logging/README.md]: Model pipeline shouldn't be the word. I was thinking something more like the camera is broken or something that causes all 0s to be sent to the model. I will update

File: examples/logging/README.md (hunk 9 above)
Comment (rsnm2) [reply examples/logging/README.md]: Which item was not implemented (enabled/disabled by default, disabling globally or at group level)? 

Could you suggest the correct verbiage?

File: examples/logging/tutorial-server-prometheus/README.md (hunk 38 above)
Comment (rsnm2) [reply examples/logging/tutorial-server-prometheus/README.md]: Sounds good, can update the image

File: examples/logging/tutorial-server-prometheus/README.md (hunk 40 above)
Comment (rsnm2) [reply examples/logging/tutorial-server-prometheus/README.md]: Yes, this is an updated version of the one you originally made. We should hold off on making the more complex one

File: examples/logging/README.md (hunk 9 above)
Comment (dbogunowicz) [reply examples/logging/README.md]: So the only thing which we add to the yaml file in this release is `logging` and `data_logging` (for endpoint). There is no functionality to change anything about system logs whatsoever. The only system logging that happens is the four latency measurements that are active by default (and cannot be turned off)

**Pull request closed without merging**